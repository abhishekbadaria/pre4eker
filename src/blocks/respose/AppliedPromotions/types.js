const types = {
  earnedBonusPoints: "Начисление баллов",
  spentBonusPoints: "Скидка балламми",
  deliveryDiscount: "Скидка на доставку",
  message: "Сообщение",
  issuedCoupon: "Выданный промокод",
  discount: "Скидка",
};

export default types;