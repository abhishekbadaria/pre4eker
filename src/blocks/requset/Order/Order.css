.inline {
  display: flex;
  width: 100%;
  justify-content: space-between;
  flex-wrap: wrap;
  margin-bottom: 16px;
}

.half {
  /* width: 50%; */
  margin-right: 8px;
  flex-grow: 1;
}

.half:last-child {
  margin-right: 0;
}