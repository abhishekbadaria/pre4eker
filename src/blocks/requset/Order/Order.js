import React, { Component } from "react";

import style from "./Order.css";
import Selector from "../Selector/Selector";
import Input from "src/blocks/Input/Input";
import CustomFields from '../CustomFields/CustomFields';

export default class Order extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedID: "mindboxId",
      customFields: []
    };

    this.ids = this.props.order.ids;

    this.changeSelection = this.changeSelection.bind(this);
    this.addCustomField = this.addCustomField.bind(this);
    this.removeCustomField = this.removeCustomField.bind(this);

  }

  changeSelection(option) {
    this.setState({
      selectedID: option,
    });
  }

  addCustomField() {
    this.setState({
      customFields:  [...this.state.customFields, 1]
    }) 
  }

  removeCustomField(e) {
 
    const array = [...this.state.customFields]; // make a separate copy of the array
  const index = array.indexOf(e.target.value)
  if (index !== -1) {
    array.splice(index, 1);
    this.setState({customFields: array});
  }
  }

  render() {
    return (
      <div>
        <h2 className={style.h2}>Заказ</h2>
        <Selector
          selected={this.state.selectedID}
          options={this.ids}
          changeSelection={this.changeSelection}
          label="Идентификатор заказа"
        />

        <div className={style.inline}>
          <div className={style.half}>
            <Input label={this.props.order.cashdesk} name="cashdesk" />
          </div>
          <div className={style.half}>
            <Input label={this.props.order.deliveryCost} name="deliveryCost" />
          </div>
        </div>
        <div className={style.inline}>
          <div className={style.half}>
            <Input label={this.props.order.area} name="area" />
          </div>
          <div className={style.half}>
            <Input
              label={this.props.order.pointOfContact}
              name="pointOfContact"
            />
          </div>
        </div>

      <CustomFields addCustomField={this.addCustomField} removeCustomField={this.removeCustomField} fields={this.state.customFields} />
      </div>
    );
  }
}
