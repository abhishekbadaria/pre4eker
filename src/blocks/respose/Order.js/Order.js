import React from 'react'
import PropTypes from 'prop-types'

const Order = ({order}) => {
  return (
    <div>
      
    </div>
  )
}

Order.propTypes = {
  order: PropTypes.object
}

export default Order
