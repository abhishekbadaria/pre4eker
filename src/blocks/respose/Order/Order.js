import React from "react";
import PropTypes from "prop-types";

import statuses from "./statuses";
import stylesShared from "src/customer.css";
import CustomFileds from "../CustomFields/CustomFields";
import Coupons from "../Coupons/Coupons";
import BonusPointsInfo from "../BonusPointsInfo/BonusPointsInfo";
import BonusPointsChange from "../BonusPointsChange/BonusPointsChange";

const Order = ({ order }) => {
  const {
    processingStatus,
    deliveryCost,
    totalPrice,
    customFields,
    couponsInfo,
    bonusPointsInfo,
    bonusPointsChanges,
    ids
  } = order;

  const drawIds = (ids) => {
    const systemNames = Object.keys(ids);
    return systemNames.map((item) =>  (<div className={stylesShared.name}>{item}: {ids[item]} </div>) )
  };

  return (
    <>
      <h2>Заказ</h2>
      <div className={stylesShared.customer}>
        <div className={stylesShared.inline}>
          <div className={stylesShared.half}>
            { drawIds(ids) }
          </div>
          <div className={stylesShared.half}>
            <div className={stylesShared.name}>Статус обработки</div>
            <div className={stylesShared.value}>
              {statuses[processingStatus] || processingStatus}
            </div>
          </div>
        </div>
        <div className={stylesShared.inline}>
          <div className={stylesShared.half}>
            <div className={stylesShared.value}>Доставка</div>
          </div>
          <div className={stylesShared.half}>
            <div className={stylesShared.value}>{deliveryCost}</div>
          </div>
        </div>
        <div className={stylesShared.inline}>
          <div className={stylesShared.half}>
            <div className={stylesShared.value}>Итоговая цена</div>
          </div>
          <div className={stylesShared.half}>
            <div className={stylesShared.value}>{totalPrice}</div>
          </div>
        </div>
        {customFields && <CustomFileds data={customFields} />}
      </div>
      {couponsInfo &&
        couponsInfo.map((coupon) => <Coupons couponInfo={coupon} />)}
      {bonusPointsInfo &&
        bonusPointsInfo.map((bonus) => (
          <BonusPointsInfo bonusPointsInfo={bonus} />
        ))}
      {bonusPointsChanges &&
        bonusPointsChanges.map((bonus) => (
          <BonusPointsChange bonusPointsChanges={bonus} />
        ))}
    </>
  );
};

Order.propTypes = {
  order: PropTypes.object,
};

export default Order;
