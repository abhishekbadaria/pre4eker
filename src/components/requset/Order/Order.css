.inline {
  display: flex;
  /* margin: 0 -.3em 0 -.3em; */
  flex-wrap: wrap;
  align-items: flex-start;
  justify-content: space-between;
  /* margin-bottom: .4rem; */
}

.half {
  width: 49%;
  /* padding: 1rem .3rem; */
}

.third {
  width: 32%;
  /* padding: 0 .3rem; */
}

.half:last-child {
  margin-right: 0;
}

.gutter {
  /* margin: 0 -.3em 0 -.3em; */
}

.cell {
  /* padding: 0 .3em 0 .3em; */
}

.h4 {
  margin: 0;
}

.line {
  margin-bottom: .5rem;
}

.description {
  font-size: .7rem;
  margin: .3rem 0 0 0;
  margin-bottom: .7rem;
  color: #777;
}

.promoItem {
  display: flex;
  /* margin: 0 -.3em 0 -.3em; */
  flex-wrap: wrap;
  align-items: flex-end;
  justify-content: space-between;
}

.promoItemInput {
  flex-shrink: 2;
}

.promoBlock {
  /* margin: .5rem 0; */
  padding: .3rem;
  width: 49%;
  align-items: center;
  border: 1px solid #E0E0E0;
}

.promoBlockHeader {
  display: flex;
  align-items: center;
}

.promoBlockHeader button {
  margin-right: .5rem;
}

.orderButtons {
  display: flex;
  font-size: .7rem;
}

.center {
  align-items: center;
}