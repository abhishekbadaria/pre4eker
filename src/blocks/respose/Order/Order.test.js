import React from "react";
import Order from "./Order";
import { render, fireEvent, waitFor, screen, cleanup, act } from '@testing-library/react' 
import response from 'Data/response'
import { exportAllDeclaration } from "@babel/types";

afterEach(cleanup)


const {order} = response;

test(`customer match snapshot`, () => {
  const {asFragment} = render(<Order order={order} />)
  expect(asFragment(<Order />)).toMatchSnapshot();

  fireEvent.click(screen.getByRole('button'));
  expect(asFragment(<Order />)).toMatchSnapshot();
  
  fireEvent.click(screen.getByRole('button'));
  expect(asFragment(<Order />)).toMatchSnapshot();
})