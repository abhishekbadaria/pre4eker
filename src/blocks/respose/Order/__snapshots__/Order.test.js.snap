// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer match snapshot 1`] = `
<DocumentFragment>
  <h2>
    Заказ
  </h2>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div>
          mindboxId: 1968413179 
        </div>
        <div>
          orderId2Id: 1784802773 
        </div>
        <div>
          orderMobileAppIdId: 1288953047 
        </div>
        <div>
          offlineTransactionIdId: 1430164306 
        </div>
        <div>
          websiteTransactionIdId: 2035457217 
        </div>
        <div>
          receiptNumberId: 107913866 
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Рассчитан
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="value"
        >
          Доставка
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="value"
        >
          20638572.96
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="value"
        >
          Итоговая цена
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="value"
        >
          10542306.11
        </div>
      </div>
    </div>
    <div>
      <div
        class="name"
      >
        Дополнительные поля
        <button
          class="small button normal shevron"
        >
          <div>
            <svg
              fill="none"
              height="5"
              viewBox="0 0 8 5"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Промокод
        </div>
        <div
          class="value"
        >
          ABC12345
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Сумма скидки
        </div>
        <div
          class="promoValue"
        >
          10528369.9
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      Пока еще не активен
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Выдан
      </div>
      <div
        class="half value"
      >
        10/23/2018
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Сгорит
      </div>
      <div
        class="half value"
      >
        11/21/2017
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Использован
      </div>
      <div
        class="half value"
      >
        4/20/2019
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Промокод
        </div>
        <div
          class="value"
        >
          ABC12345
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Сумма скидки
        </div>
        <div
          class="promoValue"
        >
          6533321.56
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      Нельзя применить на этот заказ
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Выдан
      </div>
      <div
        class="half value"
      >
        9/27/2019
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Сгорит
      </div>
      <div
        class="half value"
      >
        9/15/2017
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Использован
      </div>
      <div
        class="half value"
      >
        2/16/2018
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Информация о баллах
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Успешно
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Доступно для списание на этот заказ
        </div>
        <div
          class="promoValue"
        >
          8295431.24
        </div>
      </div>
    </div>
    <div
      class="half"
    >
      <div
        class="name"
      >
        Будет списано на этот заказ
      </div>
      <div
        class="half"
      >
        <div
          class="promoValue"
        >
          1559295.84
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="name"
      >
        Баланс
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Доступно
        </div>
        <div
          class="value"
        >
          2202060.23
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Блок
        </div>
        <div
          class="value"
        >
          19474460.28
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Всего
        </div>
        <div
          class="value"
        >
          16594789.28
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Информация о баллах
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Успешно
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Доступно для списание на этот заказ
        </div>
        <div
          class="promoValue"
        >
          15417181.89
        </div>
      </div>
    </div>
    <div
      class="half"
    >
      <div
        class="name"
      >
        Будет списано на этот заказ
      </div>
      <div
        class="half"
      >
        <div
          class="promoValue"
        >
          20423028.05
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="name"
      >
        Баланс
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Доступно
        </div>
        <div
          class="value"
        >
          8865936.3
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Блок
        </div>
        <div
          class="value"
        >
          19197363.33
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Всего
        </div>
        <div
          class="value"
        >
          15632017.52
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          P02zXwLm
        </div>
        <div
          class="status"
        >
          MBcRZtuf
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="promoValue"
        >
          1396638.62
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="promoValue"
          >
            5761799.72
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          LE0pZEmG
        </div>
        <div
          class="status"
        >
          REDNhFOU
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="promoValue"
        >
          20955125.72
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="promoValue"
          >
            10868072.91
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`customer match snapshot 2`] = `
<DocumentFragment>
  <h2>
    Заказ
  </h2>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div>
          mindboxId: 1968413179 
        </div>
        <div>
          orderId2Id: 1784802773 
        </div>
        <div>
          orderMobileAppIdId: 1288953047 
        </div>
        <div>
          offlineTransactionIdId: 1430164306 
        </div>
        <div>
          websiteTransactionIdId: 2035457217 
        </div>
        <div>
          receiptNumberId: 107913866 
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Рассчитан
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="value"
        >
          Доставка
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="value"
        >
          20638572.96
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="value"
        >
          Итоговая цена
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="value"
        >
          10542306.11
        </div>
      </div>
    </div>
    <div>
      <div
        class="name"
      >
        Дополнительные поля
        <button
          class="small button normal shevron"
        >
          <div>
            <svg
              fill="none"
              height="5"
              viewBox="0 0 8 5"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="inline"
      >
        <div
          class="half value"
        >
          defectTestField
        </div>
        <div
          class="half value"
        >
          test13
        </div>
      </div>
      <div
        class="inline"
      >
        <div
          class="half value"
        >
          socialcard
        </div>
        <div
          class="half value"
        >
          true
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Промокод
        </div>
        <div
          class="value"
        >
          ABC12345
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Сумма скидки
        </div>
        <div
          class="promoValue"
        >
          10528369.9
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      Пока еще не активен
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Выдан
      </div>
      <div
        class="half value"
      >
        10/23/2018
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Сгорит
      </div>
      <div
        class="half value"
      >
        11/21/2017
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Использован
      </div>
      <div
        class="half value"
      >
        4/20/2019
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Промокод
        </div>
        <div
          class="value"
        >
          ABC12345
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Сумма скидки
        </div>
        <div
          class="promoValue"
        >
          6533321.56
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      Нельзя применить на этот заказ
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Выдан
      </div>
      <div
        class="half value"
      >
        9/27/2019
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Сгорит
      </div>
      <div
        class="half value"
      >
        9/15/2017
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Использован
      </div>
      <div
        class="half value"
      >
        2/16/2018
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Информация о баллах
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Успешно
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Доступно для списание на этот заказ
        </div>
        <div
          class="promoValue"
        >
          8295431.24
        </div>
      </div>
    </div>
    <div
      class="half"
    >
      <div
        class="name"
      >
        Будет списано на этот заказ
      </div>
      <div
        class="half"
      >
        <div
          class="promoValue"
        >
          1559295.84
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="name"
      >
        Баланс
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Доступно
        </div>
        <div
          class="value"
        >
          2202060.23
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Блок
        </div>
        <div
          class="value"
        >
          19474460.28
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Всего
        </div>
        <div
          class="value"
        >
          16594789.28
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Информация о баллах
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Успешно
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Доступно для списание на этот заказ
        </div>
        <div
          class="promoValue"
        >
          15417181.89
        </div>
      </div>
    </div>
    <div
      class="half"
    >
      <div
        class="name"
      >
        Будет списано на этот заказ
      </div>
      <div
        class="half"
      >
        <div
          class="promoValue"
        >
          20423028.05
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="name"
      >
        Баланс
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Доступно
        </div>
        <div
          class="value"
        >
          8865936.3
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Блок
        </div>
        <div
          class="value"
        >
          19197363.33
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Всего
        </div>
        <div
          class="value"
        >
          15632017.52
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          P02zXwLm
        </div>
        <div
          class="status"
        >
          MBcRZtuf
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="promoValue"
        >
          1396638.62
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="promoValue"
          >
            5761799.72
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          LE0pZEmG
        </div>
        <div
          class="status"
        >
          REDNhFOU
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="promoValue"
        >
          20955125.72
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="promoValue"
          >
            10868072.91
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`customer match snapshot 3`] = `
<DocumentFragment>
  <h2>
    Заказ
  </h2>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div>
          mindboxId: 1968413179 
        </div>
        <div>
          orderId2Id: 1784802773 
        </div>
        <div>
          orderMobileAppIdId: 1288953047 
        </div>
        <div>
          offlineTransactionIdId: 1430164306 
        </div>
        <div>
          websiteTransactionIdId: 2035457217 
        </div>
        <div>
          receiptNumberId: 107913866 
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Рассчитан
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="value"
        >
          Доставка
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="value"
        >
          20638572.96
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="value"
        >
          Итоговая цена
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="value"
        >
          10542306.11
        </div>
      </div>
    </div>
    <div>
      <div
        class="name"
      >
        Дополнительные поля
        <button
          class="small button normal shevron"
        >
          <div>
            <svg
              fill="none"
              height="5"
              viewBox="0 0 8 5"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Промокод
        </div>
        <div
          class="value"
        >
          ABC12345
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Сумма скидки
        </div>
        <div
          class="promoValue"
        >
          10528369.9
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      Пока еще не активен
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Выдан
      </div>
      <div
        class="half value"
      >
        10/23/2018
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Сгорит
      </div>
      <div
        class="half value"
      >
        11/21/2017
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Использован
      </div>
      <div
        class="half value"
      >
        4/20/2019
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Промокод
        </div>
        <div
          class="value"
        >
          ABC12345
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Сумма скидки
        </div>
        <div
          class="promoValue"
        >
          6533321.56
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      Нельзя применить на этот заказ
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Выдан
      </div>
      <div
        class="half value"
      >
        9/27/2019
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Сгорит
      </div>
      <div
        class="half value"
      >
        9/15/2017
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half value"
      >
        Использован
      </div>
      <div
        class="half value"
      >
        2/16/2018
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Информация о баллах
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Успешно
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Доступно для списание на этот заказ
        </div>
        <div
          class="promoValue"
        >
          8295431.24
        </div>
      </div>
    </div>
    <div
      class="half"
    >
      <div
        class="name"
      >
        Будет списано на этот заказ
      </div>
      <div
        class="half"
      >
        <div
          class="promoValue"
        >
          1559295.84
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="name"
      >
        Баланс
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Доступно
        </div>
        <div
          class="value"
        >
          2202060.23
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Блок
        </div>
        <div
          class="value"
        >
          19474460.28
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Всего
        </div>
        <div
          class="value"
        >
          16594789.28
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Информация о баллах
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Статус обработки
        </div>
        <div
          class="value"
        >
          Успешно
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Доступно для списание на этот заказ
        </div>
        <div
          class="promoValue"
        >
          15417181.89
        </div>
      </div>
    </div>
    <div
      class="half"
    >
      <div
        class="name"
      >
        Будет списано на этот заказ
      </div>
      <div
        class="half"
      >
        <div
          class="promoValue"
        >
          20423028.05
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="name"
      >
        Баланс
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Доступно
        </div>
        <div
          class="value"
        >
          8865936.3
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Блок
        </div>
        <div
          class="value"
        >
          19197363.33
        </div>
      </div>
      <div
        class="third"
      >
        <div
          class="name"
        >
          Всего
        </div>
        <div
          class="value"
        >
          15632017.52
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          P02zXwLm
        </div>
        <div
          class="status"
        >
          MBcRZtuf
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="promoValue"
        >
          1396638.62
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="promoValue"
          >
            5761799.72
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="customer"
  >
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="promoType"
        >
          Изменения баллов
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Балльный счет
        </div>
        <div
          class="value"
        >
          LE0pZEmG
        </div>
        <div
          class="status"
        >
          REDNhFOU
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Начислено
        </div>
        <div
          class="promoValue"
        >
          20955125.72
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Списано
        </div>
        <div
          class="half"
        >
          <div
            class="promoValue"
          >
            10868072.91
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
